src/stores/filingHistoryListStore.ts

```diff
@@ -0,0 +1,322 @@
+    isAddCommentDialog (state): boolean {
+      return state.addCommentDialog
+    },
+
+    /** Whether the Download Error dialog should be displayed. */
+    isDownloadErrorDialog (state): boolean {
+      return state.downloadErrorDialog
+    },
+
+    /** Whether the File Correction dialog should be displayed. */
+    isFileCorrectionDialog (state): boolean {
+      return state.fileCorrectionDialog
+    },
+
+    /** Whether the Load Correction dialog should be displayed. */
+    isLoadCorrectionDialog (state): boolean {
+      return state.loadCorrectionDialog
+    },
+
+    /** Whether all documents are downloading. */
+    isLoadingAll (state): boolean {
+      return state.loadingAll
+    },
+
+    /** Whether one document is downloading. */
+    isLoadingOne (state): boolean {
+      return state.loadingOne
+    }
+  },
+
+  actions: {
+    setCurrentFiling (val: ApiFilingIF) {
+      this.currentFiling = val
+    },
+
+    setDownloadErrorDialog (val: boolean) {
+      this.downloadErrorDialog = val
+    },
+
+    setFileCorrectionDialog (val: boolean) {
+      this.fileCorrectionDialog = val
+    },
+
+    mutateFilings (val: ApiFilingIF[]) {
+      this.filings = val
+    },
+
+    setLoadingAll (val: boolean) {
+      this.loadingAll = val
+    },
+
+    setLoadCorrectionDialog (val: boolean) {
+      this.loadCorrectionDialog = val
+    },
+
+    setLoadingOne (val: boolean) {
+      this.loadingOne = val
+    },
+
+    setLoadingOneIndex (val: number) {
+      this.loadingOneIndex = val
+    },
+
+    setPanel (val: number) {
+      this.panel = val
+    },
+
+    /**
+     * Fetches the filings list for the specified id from the Legal API and,
+     * if successful, saves it in the store.
+     * @param context the Vuex context (passed in automatically)
+     * @param id a business id or temporary registration number
+     */
+    loadFilings (id: string): Promise<any> {
+    // need to return a promise
+      return new Promise((resolve, reject) => {
+        LegalServices.fetchFilings(id)
+          .then(filings => {
+            this.setFilings(filings)
+            // return the filings list
+            resolve(filings)
+          })
+          .catch(error => {
+            reject(error)
+          })
+      })
+    },
+
+    setFilings (val: ApiFilingIF[]): void {
+    // add some properties to each filing
+    // so they're not null (and therefore non-reactive)
+      const filings = val.map(filing => {
+        return {
+          ...filing,
+          comments: filing.comments || null,
+          documents: filing.documents || null
+        }
+      })
+      this.mutateFilings(filings)
+    },
+
+    /** Closes current panel or opens new panel and loads comments and documents. */
+    toggleFilingHistoryItem (index: number): Promise<void> {
+      const rootStore = useRootStore()
+      // need to return a promise because action is called via dispatch
+      return new Promise(async resolve => {
```

Comment:
[Sonarcloud reports a bug on this line.](https://sonarcloud.io/project/issues?resolved=false&types=BUG&pullRequest=500&id=bcgov_business-filings-ui&open=AYgC5YULUDaGZf7ln5iV)

Should this action actually return a `Promise<any>` ?

Reply:
Also, what part of this promise function is async?

Reply:
PS This is OK:
`new Promise(resolve => ...`

And this is OK:
`new Promise((resolve) => ...`

The reason for the second form is in case there is zero of more than 1 function parameter, eg:
`new Promise(() => ...`
or
`new Promise((resolve, reject) => ...`

Reply:
Or maybe Sonarcloud is complaining about "async resolve" and it would prefer "async (resolve)" ?

Reply:
I'd like your thoughts on the new version of this Sev. SonarCloud was complaining a lot about this and labelling it as a bug ("Promise returned in function argument where a void return was expected").

I made some research and ended up not returning a Promise anymore. Let me know what do you think of the change.

Reply:
You know, I never really liked the returned promise -- I thought it was redundant. It looks like you figured it out! I ran your code locally and the filing history item opens, and the comments and documents load just fine, and no console messages.

LGTM!